Tasks/FtpUpload/ftpuploadtask.ts

```diff
@@ -0,0 +1,278 @@
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+/// <reference path="../../definitions/node.d.ts" />
+/// <reference path="../../definitions/Q.d.ts" />
+
+import fs = require('fs');
+import os = require('os');
+import path = require('path');
+import tl = require('vsts-task-lib/task');
+import url = require('url');
+
+var SortedSet = require('collections/sorted-set');
+var Client = require('ftp');
+
+var win = os.type().match(/^Win/);
+tl.debug('win: ' + win);
+
+var repoRoot: string = tl.getVariable('build.sourcesDirectory');
+function makeAbsolute(normalizedPath: string): string {
```

Comment:
doesn't look like that has made it in yet, once it's there, I can go back and fix.